File: medicycle/fix-admin-imports.js
const fs = require('fs');
const path = require('path');

const walkSync = (dir, filelist = []) => {
  fs.readdirSync(dir).forEach(file => {
    const dirFile = path.join(dir, file);
    if (fs.statSync(dirFile).isDirectory()) {
      filelist = walkSync(dirFile, filelist);
    }
    else if (dirFile.endsWith('.tsx') || dirFile.endsWith('.ts')) {
      filelist.push(dirFile);
    }
  });
  return filelist;
};

const adminDir = '/Users/ravendrapatel/Documents/Buyer seller new /medicycle/apps/admin/src';
const files = walkSync(adminDir);

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let originalContent = content;
  
  content = content.replace(/from "\.\.\/\.\.\/design-system\/utils\/cn"/g, 'from "@medicycle/utils"');
  content = content.replace(/from "\.\.\/\.\.\/\.\.\/design-system\/utils\/cn"/g, 'from "@medicycle/utils"');
  content = content.replace(/from "\.\.\/\.\.\/design-system\/components\/Button\/Button"/g, 'from "@medicycle/ui"');
  content = content.replace(/from "\.\.\/\.\.\/\.\.\/design-system\/components\/Button\/Button"/g, 'from "@medicycle/ui"');
  content = content.replace(/from "\.\.\/\.\.\/lib\/seo\/SEO"/g, 'from "../lib/seo/SEO"');
  content = content.replace(/from "\.\.\/components\/common\/LoadingScreen"/g, 'from "@medicycle/ui"');
  content = content.replace(/from "\.\.\/\.\.\/components\/common\/LoadingScreen"/g, 'from "@medicycle/ui"');
  content = content.replace(/import { ThemeToggle } from "\.\.\/\.\.\/\.\.\/components\/common\/ThemeToggle"/g, 'import { ThemeToggleWrapped as ThemeToggle } from "./ThemeToggleWrapper"');
  
  // Specific fix for AdminRouteGuard LoadingScreen
  if (file.includes('AdminRouteGuard')) {
      content = content.replace(/from "\.\.\/\.\.\/components\/common\/LoadingScreen"/g, 'from "@medicycle/ui"');
  }

  if (content !== originalContent) {
    fs.writeFileSync(file, content, 'utf8');
    console.log(`Updated ${file}`);
  }
});

File: medicycle/fix-admin-imports.py
import os
import re

admin_dir = '/Users/ravendrapatel/Documents/Buyer seller new /medicycle/apps/admin/src'

patterns = [
    (r'from [\'"]\.\./\.\./design-system/utils/cn[\'"]', 'from "@medicycle/utils"'),
    (r'from [\'"]\.\./\.\./\.\./design-system/utils/cn[\'"]', 'from "@medicycle/utils"'),
    (r'from [\'"]\.\./\.\./design-system/components/Button/Button[\'"]', 'from "@medicycle/ui"'),
    (r'from [\'"]\.\./\.\./\.\./design-system/components/Button/Button[\'"]', 'from "@medicycle/ui"'),
    (r'from [\'"]\.\./\.\./lib/seo/SEO[\'"]', 'from "../lib/seo/SEO"'),
    (r'from [\'"]\.\./components/common/LoadingScreen[\'"]', 'from "@medicycle/ui"'),
    (r'from [\'"]\.\./\.\./components/common/LoadingScreen[\'"]', 'from "@medicycle/ui"'),
    (r'import \{ ThemeToggle \} from [\'"]\.\./\.\./\.\./components/common/ThemeToggle[\'"]', 'import { ThemeToggleWrapped as ThemeToggle } from "./ThemeToggleWrapper"')
]

for root, dirs, files in os.walk(admin_dir):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            original_content = content
            for pattern, repl in patterns:
                content = re.sub(pattern, repl, content)
                
            if 'AdminRouteGuard' in path:
                content = re.sub(r'from [\'"]\.\./\.\./components/common/LoadingScreen[\'"]', 'from "@medicycle/ui"', content)
                
            if content != original_content:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f'Updated {path}')
